pr_data_wrangling.py

```diff
@@ -0,0 +1,154 @@
+
+
+### dictionary to collapse the subspecimens
+## techniques/modalities/species will be collapsed in a similar manner until this gets integrated with ontology at a future date TBD
+specimen_count_map = {
+'brain_count' : ['whole brain',
+            'brain', 'brains'], # fix the plural
+'cell_count' : ['whole cell', 
+           'cell', 'cells', 'cell body', 'cell nucleus', 'nuclei', 'reconstruction', 'cell in slice'],
+'nucleus_count' : ['cell nucleus', 'nuclei'],
+'library_count' : ['library'],
+'tissue_region_count' : ['brain hemisphere', 
+                    'brain region', 
+                    'brain section set', 
+                    'brain slice', 
+                    'library', 
+                    'spinal cord slice',
+                    'tissue sample',
+                    'cell suspension']
+}
+    
+    
+## collapse using dictionary  
+count_df = pd.DataFrame() # container to store
+    
+# for each project
+for project in df_donor_counts['data_collection_reference_id'].unique():
+    # list of column names to keep
+    non_count_cols = ['data_collection_reference_id', 'grant_reference_id', 'quarters', 'species', 'donor_count', 'subspecimen_type']
+    # use the above column names and the dictionary keys together as column names of the count dataframe
+    count_df_project = pd.DataFrame(columns = [non_count_cols + list(specimen_count_map.keys())], index = [0])
+
+    project_df = df_donor_counts[df_donor_counts['data_collection_reference_id'] == project]
+    # if there are rows with same information, but different counts, sum those counts
+    project_df = project_df.groupby(non_count_cols, as_index = False)['subspecimen_count'].sum() 
+
+    ## for each quarter
+    # could potentially write this into a function
+    for i in range(len(project_df)):
+        project_row = project_df.iloc[i]
+        row_container = pd.DataFrame(columns = [non_count_cols + list(specimen_count_map.keys())], index = [0]) # another container
+        #row_container[non_count_cols] = project_row[non_count_cols].drop('subspecimen_type')
+        # copy the metadata
+        row_container[['data_collection_reference_id', 'grant_reference_id', 'quarters', 'species', 'donor_count']] = project_row[['data_collection_reference_id', 'grant_reference_id', 'quarters', 'species', 'donor_count']]
+        # get the categories to count, based on the dictionary
+        count_cols = [k for k, v in specimen_count_map.items() if project_row['subspecimen_type'] in v]
+        # update the according column
+        row_container[count_cols] = project_row['subspecimen_count']
+        # concat to container
+        count_df_project = pd.concat([count_df_project, row_container.fillna(0)])
+    
+    count_df_project = count_df_project.dropna(how = 'all')
+    count_df = pd.concat([count_df, count_df_project])
+
+## output as csv for dashboard input
+count_df.to_csv(os.path.join(cwd, 'dash_count_df.csv'), index = False)
```

Comment:
It is convention to add newlines at the end of all files: https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

This auto-formats to:
```suggestion
count_df.to_csv(os.path.join(cwd, "dash_count_df.csv"), index=False)

```